internal/net/http/transport/roundtrip.go

```diff
@@ -41,50 +42,66 @@ func NewExpBackoff(opts ...Option) http.RoundTripper {
 	return e
 }
 
-func (e *ert) RoundTrip(req *http.Request) (res *http.Response, err error) {
+// RoundTrip round trip the HTTP request and return the response.
+// If backoff is not set, the default roundTrip implementation will be used.
+// The default roundTrip implementation round trip the request, and return any round trip error occurred.
+// It returns errors.ErrTransportRetryable to indicate if the request is consider as retryable.
+func (e *ert) RoundTrip(req *http.Request) (*http.Response, error) {
 	if e.bo == nil {
 		return e.roundTrip(req)
 	}
-	var r interface{}
-	r, err = e.bo.Do(req.Context(), func() (interface{}, error) {
+
+	r, err := e.bo.Do(req.Context(), func() (interface{}, error) {
 		return e.roundTrip(req)
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	var ok bool
-	res, ok = r.(*http.Response)
+	res, ok := r.(*http.Response)
 	if !ok {
 		return nil, errors.ErrInvalidTypeConversion(r, res)
 	}
 	return res, nil
 }
 
-func (e *ert) roundTrip(req *http.Request) (res *http.Response, err error) {
-	res, err = e.transport.RoundTrip(req)
-	if err == nil {
-		return res, nil
-	}
-	if res == nil {
-		return nil, err
-	}
-	_, err = io.Copy(ioutil.Discard, res.Body)
+func (e *ert) roundTrip(req *http.Request) (*http.Response, error) {
+	res, err := e.transport.RoundTrip(req)
 	if err != nil {
-		log.Error(err)
+		if res != nil { // just in case we check the response as it depends on RoundTrip impl.
+			closeBody(res)
+			if retryable(res) {
+				return nil, errors.ErrTransportRetryable
+			}
```

Comment:
🤔 
```suggestion
			}
			return res, err
```

Reply:
I think returning `nil` instead of `res` is better 🤔
what do you think?

Reply:
we already returns error at L77

Reply:
@kpango 
I'm sorry. I was reading it wrongly.